### PR TITLE
feat: Separate key

### LINKED ISSUE #113: feat(register): How to register the public key
## Description

If we register the user from the web dashboard, then how to provide the ssh public key

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by gaocegege**
Close #113 

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

File: pkg/service/user/sshkey.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+func (u *generalService) CreatePubKey(ctx context.Context, loginName, name string, pubKey []byte) error {
+	_, err := u.querier.CreateKey(ctx, query.CreateKeyParams{
+		LoginName: loginName,
+		PublicKey: pubKey,
+	})
+	if err != nil {
+		var pgErr *pgconn.PgError
+		if errors.As(err, &pgErr) {
+			switch pgErr.Code {
+			case "23505":
```
Comment (VoVAllen) [inline pkg/service/user/sshkey.go:30]: What's this magic number?

**Review by VoVAllen — COMMENTED** [review]
Other lgtm

File: pkg/service/user/sshkey.go (hunk 1 above)
Comment (VoVAllen) [reply pkg/service/user/sshkey.go:30]: They have an error code lib https://github.com/jackc/pgerrcode/blob/master/errcode.go

Comment (gaocegege) [reply pkg/service/user/sshkey.go:30]: OK let's use it.

**Pull request merged**